sdk/resourcemanager/Azure.ResourceManager.Core/src/GenericResourceOperations.cs

```diff
@@ -233,10 +234,29 @@ public async Task<ArmOperation<GenericResource>> StartRemoveTagAsync(string key,
         {
             GenericResource resource = await GetResourceAsync(cancellationToken).ConfigureAwait(false);
             resource.Data.Tags.Remove(key);
-            var op = await Operations.StartUpdateByIdAsync(Id, await ClientOptions.ApiVersions.TryGetApiVersionAsync(Id.Type, cancellationToken).ConfigureAwait(false), resource.Data, cancellationToken).ConfigureAwait(false);
+            var op = await Operations.StartUpdateByIdAsync(Id, await GetApiVersionAsync(cancellationToken).ConfigureAwait(false), resource.Data, cancellationToken).ConfigureAwait(false);
             return new PhArmOperation<GenericResource, ResourceManager.Resources.Models.GenericResource>(
                 await op.WaitForCompletionAsync(cancellationToken).ConfigureAwait(false),
                 v => new GenericResource(this, new GenericResourceData(v)));
         }
+
+        private string GetApiVersion(CancellationToken cancellationToken)
+        {
+            string version = ClientOptions.ApiVersions.TryGetApiVersion(Id.Type, cancellationToken);
+            if (version == null)
+            {
+                throw new ArgumentException($"An invalid resouce id was given {Id}");
+            }
+            return version;
+        }
+        private async Task<string> GetApiVersionAsync(CancellationToken cancellationToken)
+        {
+            string version = await ClientOptions.ApiVersions.TryGetApiVersionAsync(Id.Type, cancellationToken).ConfigureAwait(false);
+            if (version == null)
```

Comment:
use is to check null
```suggestion
            if (version is null)
```